yesses/discover/webservers.py

```diff
@@ -78,24 +95,30 @@ def run(self):
         output_secure = []
         tls_domains = []
         for ip in self.ips:
-            for domain in self.domains:
-                for protocol in ('http', 'https'):
-                    with force_ip_connection(domain, ip):
-                        url = f'{protocol}://{domain}/'
-                        try:
-                            result = requests.get(url, timeout=10)
-                        except requests.exceptions.RequestException as e:
-                            log.debug(f"Exception {e} on {url}, ip={ip}")
-                        else:
-                            el = {'url': url, 'domain': domain, 'ip': ip}
-                            if protocol == 'https':
-                                output_secure.append(el)
-                                domain = {'domain': domain}
-                                if not domain in tls_domains:
-                                    tls_domains.append(domain)
+            # just check a port if it is open and in the list of passed ports
+            if ip['port'] in self.ports:
+                for domain in self.domains:
+                    for protocol in ('http', 'https'):
+                        with force_ip_connection(domain, ip['ip']):
+                            url = f"{protocol}://{domain}:{ip['port']}/"
+                            try:
+                                result = requests.get(url, timeout=10)
+                                # Don't add it as a web server if it just redirects the url
+                                # or doesn't respond with 200
+                                if url not in result.url or result.status_code != 200:
```

Comment:
This does not work as intended: May webserver will redirect immediately to somewhere else. This can or cannot be a security problem as well, so it should not be filtered out.
Same for the status code. If a status code is returned, a web server is active. If anything, we could add an input parameter "ignore_error_responses" that, if set to True, would ignore any responses with 4xx and 5xx status codes.

Reply:
Ok i have changed this now. The reason why i filtered it was because the linked_path module would search the same web server several times. But i understand that i have to filter it in the linked_path module.